sqlmesh/core/dialect.py

```diff
@@ -658,11 +661,23 @@ def _is_command_statement(command: str, tokens: t.List[Token], pos: int) -> bool
         return False
 
 
+MACRO_BEGIN = "MACRO_BEGIN"
```

Comment:
I think this should just be `MACRO (...)`

Reply:
Yes and it was to avoid collisionswith SQL dialects that use MACRO statements

Reply:
i think it should be MACRO() since a macro can only have one statement.

what dialects have a macro keyword?

Reply:
macro's shouldn't require custom parsing, it can probably be something like

```@def(MY_MACRO, arg -> ...)```

Reply:
Duckdb has macro statements that use the macro keyword. Ok, I will try with a syntax like: `@def(MY_MACRO, arg -> ...)` so that it is like the sqlmesh macro definitions in the models sql files